File: solutions/go/bird-watcher/2/bird_watcher.go
package birdwatcher

func TotalBirdCount(birdsPerDay []int) int {
	s := 0
    for _, v := range(birdsPerDay) {
        s += v
    }
    return s
}

func BirdsInWeek(birdsPerDay []int, week int) int {
    s := 0
	for _, v := range(birdsPerDay[7 * (week-1):7 * week]) {
        s += v
    }
    return s
}

func FixBirdCountLog(birdsPerDay []int) []int {
	for i := range(birdsPerDay) {
        if i % 2 == 1 {
            continue
        }
        birdsPerDay[i]++
    }
    return birdsPerDay
}

File: solutions/javascript/bird-watcher/2/bird-watcher.js
export function totalBirdCount(birdsPerDay) {
  return birdsPerDay.reduce((a, b) => a + b, 0)
}

export function birdsInWeek(birdsPerDay, week) {
  return birdsPerDay.slice(7 * (week-1), 7 * week).reduce((a, b) => a + b, 0)
}

export function fixBirdCountLog(birdsPerDay) {
  birdsPerDay.forEach((v, i, a) => a[i] = i % 2 === 0 ? v + 1 : v)
  return birdsPerDay
}